locale/inc/leatherman/locale/locale.hpp

```diff
@@ -4,13 +4,91 @@
 */
 #pragma once
 #include <locale>
+#include <vector>
+#include <boost/locale/format.hpp>
 
 namespace leatherman { namespace locale {
 
     /**
      * Gets a locale object for the specified locale id.
      * @param id The locale ID, defaults to a UTF-8 compatible system default.
+     * @param domain The catalog domain to use for i18n via gettext.
+     * @param paths Search paths for localization files.
+     * @return The locale. If a locale for the specified domain already exists, it returns
+     * the same locale until clear_domain is called for that domain.
+     * Throws boost::locale::conv::conversion_error if the system locale is invalid or
+     * the catalog for the specified language can't be used with the system locale encoding.
      */
-    std::locale get_locale(std::string const& id = "");
+    const std::locale& get_locale(std::string const& id = "", std::string const& domain = PROJECT_NAME, std::vector<std::string> const& paths = {PROJECT_DIR});
```

Comment:
Add a warning about locale ref lifetime (until another project calls get_locale) or change to using an iterator preserving data structure.